src/gui/folderman.cpp

```diff
@@ -129,28 +122,19 @@ void FolderMan::unloadFolder(Folder *f)
         &f->syncEngine().syncFileStatusTracker(), &SyncFileStatusTracker::slotPathTouched);
 }
 
-int FolderMan::unloadAndDeleteAllFolders()
+void FolderMan::unloadAndDeleteAllFolders()
 {
-    int cnt = 0;
-
     // clear the list of existing folders.
-    Folder::MapIterator i(_folderMap);
-    while (i.hasNext()) {
-        i.next();
-        Folder *f = i.value();
-        unloadFolder(f);
-        delete f;
-        cnt++;
+    const auto folders = std::move(_folderMap);
```

Comment:
Why don't you use qAsConst that clearly shows your intention?

Reply:
I clear the _folderMap with that call and unregister it.
The old code remove the folder in each iteration and made sure the list was empty afterwards...